components/net/image_cache.rs

```diff
@@ -390,21 +391,22 @@ impl ImageCache for ImageCacheImpl {
         }
     }
 
-    /// Return any available metadata or image for the given URL,
-    /// or an indication that the image is not yet available if it is in progress,
-    /// or else reserve a slot in the cache for the URL if the consumer can request images.
-    fn find_image_or_metadata(
+    fn get_image(&self, url: &ServoUrl, use_placeholder: UsePlaceholder) -> Option<Arc<Image>> {
```

Comment:
Rather than having a `use_placeholder` argument, we should have two methods - one named `get_available_image` that returns `Option<Arc<Image>>`, and one named `get_image_or_placeholder` that returns `Arc<Image>`.

Reply:
That being said, if there are no callers passing UsePlaceholder::Yes after addressing my comments, then this would be unnecessary and we could just remove the argument.

Reply:
Word, the two functions definitely makes more sense. Checking the completed loads for the loaded image or the placeholder image looks like it can still return an `Err` in the event that for whatever reason, neither image could load. `ImageCacheStore:placeholder_image` is also wrapped in an `Option`. How do I confidently return an `Arc<Image>` for `get_image_or_placeholder`? Can I just `unwrap` the `ImageCacheStore:placeholder_image` if `get_image_or_placeholder` fails? i.e. is the placeholder confidently loaded somewhere else, since I see `.ok()` is called on it when creating the store?

Reply:
We already infallibly load the placeholder image in [this code](https://github.com/servo/servo/blob/df73c71fb18c144abedd6889c34dd1031ffcbb8e/components/net/image_cache.rs#L44-L51) - we should make its return type `Arc<Image>` and update the `placeholder_image` member to store `Arc<Image>` instead of an optional value.

Reply:
Looks like nothing is using `UsePlaceholder::Yes` so doesn't matter anymore - will remove arg